src/instructions/port.rs

```diff
@@ -198,35 +180,27 @@ impl<T: PortWrite> Port<T> {
     }
 }
 
-macro_rules! impl_port_util_traits {
-    ($struct_name:ident) => {
-        impl<T> fmt::Debug for $struct_name<T> {
-            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                f.debug_struct(stringify!($struct_name))
-                    .field("port", &self.port)
-                    .finish()
-            }
-        }
-
-        impl<T> Clone for $struct_name<T> {
-            fn clone(&self) -> Self {
-                Self {
-                    port: self.port,
-                    phantom: PhantomData,
-                }
-            }
-        }
+impl<T, A> fmt::Debug for PortGeneric<T, A> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PortGeneric")
+            .field("port", &self.port)
+            .finish()
```

Comment:
Perhaps we could include the access type here as well, e.g. as an additional field? What do you think about this, @josephlr?

Reply:
The access level seems like useful info to have in the debug output, I like that idea.

Reply:
So I played around with ways to add an "access" field to the Debug representation, and it's a little messy: https://github.com/rust-osdev/x86_64/commit/bbb049362d6f69ec7edb3c97d16d47d5e62ea683

But it produces nice output:
```
PortGeneric {
    port: 80,
    size: 1,
    access: ReadWrite,
}
```

An alternative to this messy implementation is to just use const generics

Reply:
So using const generics doesn't seem too bad, I put together #254 and it looks reasonable.

The debug output looks like:
```
PortGeneric {
    port: 80,
    size: 1,
    read: true,
    write: true,
}
```